src/librustc/ty/context.rs

```diff
@@ -481,6 +481,8 @@ pub struct GlobalCtxt<'tcx> {
 
     pub maybe_unused_trait_imports: NodeSet,
 
+    pub maybe_unused_extern_crates: Vec<(NodeId, Span, CrateNum)>,
```

Comment:
Could you also add the `pub extern crate` case to tests for this lint (`compile-fail/lint-unused-extern-crate.rs`)?